portal/models/role.py

```diff
@@ -86,10 +86,7 @@ def display_name(self):
 
 
 ROLE = Enum('ROLE', {r.upper(): r for r in STATIC_ROLES})
-ALL_BUT_WRITE_ONLY = Enum(
-    'ALL_BUT_WRITE_ONLY',
-    {r.upper(): r for r in STATIC_ROLES if r != 'write_only'})
-
+ALL_BUT_WRITE_ONLY = [r.value for r in ROLE if r.value != 'write_only']
```

Comment:
Why not an immutable type eg set/tuple?

Reply:
Unfortunate design of the flask-user `@roles.required` API into which this variable is passed.  It interprets a `tuple` as a list for which the user must have *all* of the roles, and a `list` as a list of options for which the user must have at least one.  We need the later, so it must be a list.